### PR TITLE
Switch To OpenTelemetry From MSCV

### PR DESCRIPTION
**Pull request by alfredoihernandez**
image

Closes: https://github.com/retaildevcrews/wcnp/issues/626

### CODE REVIEW

File: src/Program.cs [hunk 1]
```diff
@@ -161,6 +166,21 @@ private static IWebHost BuildHost()
                     }
                 });
 
+            string serviceName = "Ngsa.Application";
+            string serviceVersion = "1.1.0";
+            builder.ConfigureServices(services => services.AddOpenTelemetryTracing(b =>
```
Comment (kforeverisback) [inline src/Program.cs:171]: [Optional] Is it possible to lump ConfigureService call to previous statement in line 151?

```C#
WebHohst.CreateDefaultBuilder()
    .UseUrls()
    ...
    ...
    .ConfigureLogging()
    ...
    .ConfigureServices(service => services.AddOpenTelemetryTracing(...) ...)
```

This is just to be consistent with the rest of the code.

File: src/Program.cs [hunk 2]
```diff
@@ -161,6 +166,21 @@ private static IWebHost BuildHost()
                     }
                 });
 
+            string serviceName = "Ngsa.Application";
+            string serviceVersion = "1.1.0";
```
Comment (kforeverisback) [inline src/Program.cs:170]: Should this be static?

Comment (alfredoihernandez) [reply src/Program.cs:170]: actually i forgot i should pull this in from assembly file instead. will fix

File: src/Controllers/DataService.cs [hunk 3]
```diff
@@ -48,17 +48,10 @@ public static async Task<IActionResult> Read<T>(HttpRequest request)
 
             string path = RequestLogger.GetPathAndQuerystring(request);
 
-            CorrelationVector cVector = Middleware.CorrelationVectorExtensions.GetCorrelationVectorFromContext(request.HttpContext);
```
Comment (kforeverisback) [inline src/Controllers/DataService.cs:51]: This looks like we are sending Correlation Vector to CosmosDB in the HTTP header. Does this have any significance for tracking CosmosDB requests? We should confirm with Joseph or Bart.

Also, should we consider add B3-tracing in place of CV or removing is fine?

Comment (alfredoihernandez) [reply src/Controllers/DataService.cs:51]: opentelemetry autoinstrumentation is replacing the removed code

**Pull request closed without merging**